src/renderers/shared/server/ReactPartialRenderer.js

```diff
@@ -484,6 +484,19 @@ class ReactDOMServerRenderer {
       if (child === null || child === false) {
         return '';
       } else {
+        if (Array.isArray(child)) {
+          var frame = {
+            children: child,
+            childIndex: 0,
+            context: context,
+            footer: '',
+          };
+          if (__DEV__) {
+            frame.debugElementStack = [];
```

Comment:
Do warnings still work? You can add a test similar to [this one](https://github.com/facebook/react/blob/240b84ed8e1db715d759afaae85033718a0b24e1/src/renderers/dom/shared/__tests__/ReactDOMComponent-test.js#L1651) but using an array.